desktop/apps/consignments2/components/describe_work/index.js

```diff
@@ -1,33 +1,176 @@
+        <div className={b('row-item')}>
+          <Field name='depth' component={renderTextInput}
+            item={'depth'}
+            label={'Depth'}
+          />
+        </div>
+        <div className={b('row-item')}>
+          <Field name='dimensions_metric' component={renderSelectInput}
+            item={'dimensions_metric'}
+            label={'Units'}
+            options={['in', 'cm']}
+          />
+        </div>
+      </div>
+      <div className={b('row', {'border-bottom': true})}>
+        <div className={b('row-item')}>
+          <Field name='edition' component={renderCheckboxInput}
+            item={'edition'}
+            label={'This is an edition'}
+          />
+        </div>
+      </div>
+      <div className={b('small-row', {'border-bottom': true})}>
+        <div className={b('row-item')}>
+          <Field name='signature' component={renderRadioInput}
+            item={'signature'}
+            label={'Is this work signed?'}
+            options={['yes', 'no']}
+          />
+        </div>
+      </div>
+      <div className={b('small-row', {'border-bottom': true})}>
+        <div className={b('row-item')}>
+          <Field name='authenticity_certificate' component={renderRadioInput}
+            item={'authenticity_certificate'}
+            label={'Does this work come with a certificate of authenticity?'}
+            options={['yes', 'no']}
+          />
+        </div>
+      </div>
+      <div className={b('row')}>
+        <div className={b('row-item')}>
+          <Field name='provenance' component={renderTextInput}
+            item={'provenance'}
+            instructions={'Where did you acquire this work?'}
+            label={'Provenance'}
+          />
+        </div>
+      </div>
+      <div className={b('row')}>
+        <div className={b('row-item')}>
+          <Field name='location' component={renderTextInput}
+            item={'location'}
+            instructions={'What city is the work located in?'}
+            label={'Location'}
+          />
+        </div>
+      </div>
+      <button
         className={b('next-button').mix('avant-garde-button-black')}
-        onClick={incrementStepAction}
+        disabled={pristine || invalid}
+        type='submit'
       >
         Next
-      </div>
-    </div>
+      </button>
+    </form>
   )
 }
 
 const mapDispatchToProps = {
-  incrementStepAction: incrementStep
+  submitDescribeWorkAction: submitDescribeWork
+}
+
+DescribeWork.propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+  invalid: PropTypes.bool,
+  pristine: PropTypes.bool,
+  submitDescribeWorkAction: PropTypes.func.isRequired
 }
 
-export default connect(
+DescribeWork = connect(
   null,
   mapDispatchToProps
 )(DescribeWork)
 
-DescribeWork.propTypes = {
-  incrementStepAction: PropTypes.func.isRequired
-}
+DescribeWork = reduxForm({
+  form: 'describeWork', // a unique identifier for this form
+  validate
+})(DescribeWork)
+
+export default DescribeWork
```

Comment:
This is an elegant functional alternative to constantly mutating the component as it's wrapped with HOC's:

```javascript
import { compose } from 'underscore'
import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'

function MyComponent() {
  ...
}

export default compose(
  reduxForm({
    foo: 'bar',
    ...
  }),
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(MyForm)
```

Compose takes `n` number of parameters functions and returns a function waiting to be called, and what it's called with is then "threaded" through each function. Since both `connect` and `reduxForm` are HOC's that are meant to return a component (and thus possess identical signatures), `MyComponent` is "decorated" as it passes through each little portal. (Also note that depending on what your composing together, order sometimes matters, just not in this case.)